winch/codegen/src/visitor.rs

```diff
@@ -205,6 +305,25 @@ where
     wasmparser::for_each_operator!(def_unsupported);
 }
 
+impl<'a, M> CodeGen<'a, M>
+where
+    M: MacroAssembler,
+{
+    fn cmp_i32s(&mut self, kind: CmpKind) {
+        self.context
+            .i32_binop(self.masm, &mut |masm, dst, src, size| {
+                masm.cmp(dst, dst, src, kind, size);
+            });
+    }
+
+    fn cmp_i64s(&mut self, kind: CmpKind) {
+        self.context
+            .i64_binop(self.masm, &mut move |masm, dst, src, size| {
+                masm.cmp(dst, dst, src, kind, size);
+            });
+    }
```

Comment:
I opted to create helper functions because the only differences between the various instruction implementations is whether they're 32 bit or 64 bit and the kind of comparison that's being performed. Everything else is the same and having it inlined everywhere IMO makes adds noise around the things that are different. I could make these macros instead but functions seem to be sufficient for the outcome I wanted and seem easier to maintain to me.